pkg/skaffold/build/buildpacks/lifecycle.go

```diff
@@ -48,17 +50,25 @@ func (b *Builder) build(ctx context.Context, out io.Writer, a *latest.Artifact,
 
 	builderImage := artifact.Builder
 	logrus.Debugln("Builder image", builderImage)
-	if err := b.pull(ctx, out, builderImage, artifact.ForcePull); err != nil {
-		return "", err
+	if !artifact.ForcePull {
+		if err := b.pull(ctx, out, builderImage); err != nil {
+			return "", err
+		}
 	}
 
-	runImage, err := b.findRunImage(ctx, artifact, builderImage)
-	if err != nil {
-		return "", err
-	}
-	logrus.Debugln("Run image", runImage)
-	if err := b.pull(ctx, out, runImage, artifact.ForcePull); err != nil {
-		return "", err
+	runImage := artifact.RunImage
+	if !artifact.ForcePull {
+		// If ForcePull is true, we let pack find and pull the run image
+		var err error
+		runImage, err = b.findRunImage(ctx, artifact, builderImage)
```

Comment:
Doesn't pack already do this?

Reply:
pack is either pulling always or pulling never. This is trying to `pullIfNotPresent`. I'll update the code with a comment.